File: src/main/java/com/goudourasv/courses/controller/dto/CourseUpdate.java
package com.goudourasv.courses.controller.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotBlank;


public class CourseUpdate {
    private String title;
    @JsonProperty("institution")
    private String institutionName;
    private String tag;
    private String professor;

    public String getTitle() {
        return title;
    }

    public String getInstitutionName() {
        return institutionName;
    }

    public String getTag() {
        return tag;
    }

    public String getProfessor() {
        return professor;
    }

}

File: src/main/java/com/goudourasv/http/courses/dto/CourseUpdate.kt
package com.goudourasv.http.courses.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.goudourasv.domain.lectures.Lecture
import com.goudourasv.domain.lectures.LectureData
import java.time.Instant
import java.util.*


data class CourseUpdate(
    val title: String?,
    val startDate: Instant?,
    val endDate: Instant?,
    @JsonProperty("institution")
    val institutionId: UUID?,
    @JsonProperty("instructor")
    val instructorId: UUID?,
    val tags: MutableSet<String>? = mutableSetOf(),
    @JsonProperty("lectures")
    val lectures: List<LectureData>? = mutableListOf(),

)